src/crawlers.ts

```diff
@@ -0,0 +1,138 @@
+import { Actor, RequestQueue } from 'apify';
+import { CheerioAPI } from 'cheerio';
+import type { RequestOptions } from 'crawlee';
+import {
+    CheerioCrawlingContext,
+    log,
+    PlaywrightCrawlingContext,
+    CheerioCrawler,
+    PlaywrightCrawler,
+    PlaywrightCrawlerOptions,
+} from 'crawlee';
+import { ServerResponse } from 'http';
+
+import { scrapeOrganicResults } from './google-extractors-urls';
+import { genericHandler } from './playwright-req-handler.js';
+import { createResponse } from './responses.js';
+import { ScraperSettings, UserData, CrawlerOptions } from './types.js';
+import { createRequest } from './utils.js';
+
+enum CrawlerType {
+    CHEERIO_GOOGLE_SEARCH_CRAWLER = 'cheerio-google-search-crawler',
+    PLAYWRIGHT_CONTENT_CRAWLER = 'playwright-content-crawler',
+}
+
+const crawlers = new Map<string, CheerioCrawler | PlaywrightCrawler>();
+const queueSearchCrawler = await RequestQueue.open('cheerio-google-search-queue');
+
+// export const DEFAULT_CRAWLER_OPTIONS: CrawlerOptions = {
+//     proxyConfigurationOptions: {},
+// };
+//
+// export const DEFAULT_SCRAPER_SETTINGS: ScraperSettings = {
+//     dynamicContentWaitSecs: 0,
+//     maxHtmlCharsToProcess: 1.5e6,
+//     readableTextCharThreshold: 100,
+//     removeCookieWarnings: true,
+//     saveHtml: true,
+//     saveMarkdown: false,
+// };
+
+// Use crawlOptions to create a new CheerioCrawler (proxyConfiguration)
+// Proxy configuration for a PlaywrightCrawler
+// How to handle the case when all the requests are not finished?
+
+export async function createAndStartSearchCrawler(
+    crawlerOptions: CrawlerOptions = {} as CrawlerOptions,
+    scraperSettings: ScraperSettings = {} as ScraperSettings,
+    startCrawler: boolean = true,
+) {
+    const proxyConfig = await Actor.createProxyConfiguration(crawlerOptions.proxyConfigurationOptions);
+
```

Comment:
Is there a reason you are not using the [memory storage as in the SuperScraper](https://github.com/apify/super-scraper/blob/master/src/crawlers.ts#L19)? It is a tiny bit faster, you don't need to use a named queue (since the queues don't collide) that you would need to delete. We use memory storage because after migration you loose the connection anyway so users need to resend the request, there is no need to persist anything.

Reply:
Good point! I had tried separating the queues earlier, but for some reason, they were colliding, and I couldn't figure out why.

Here's the relevant code:
```
    const client = new MemoryStorage();
    const queue = await RequestQueue.open(undefined, { storageClient: client });

    const crawler = new CheerioCrawler({
        ...(cheerioCrawlerOptions as CheerioCrawlerOptions),
        requestQueue: queue,
```
and
```
    const crawler = new PlaywrightCrawler({
        ...(crawlerOptions as PlaywrightCrawlerOptions),
        requestQueue: await RequestQueue.open(),
```
The issue is that the second queue ends up with the same ID and the same client, causing the queues to collide.

@metalwarrior665 Can you please help with that (there must be some fundaments I'm missing). I added it into this branch [feat/replace-queue](https://github.com/apify/rag-web-browser/blob/feat/replace-queue/src/crawlers.ts)

Reply:
I was debugging this and it seems the only way to make this work is to make both queues with memory storage AND gives them different names.

Reply:
Also add `
const client1 = new MemoryStorage({ persistStorage: false});`

Reply:
Thanks, it is working fine. Done